### DISCUSSION TITLE
Call to a member function sync() on null

### DISCUSSION THREAD

**Discussion by ravitninjatech** [spatie/laravel-permission#2450] [Q&A] [answered]
when i assign role to user that time i got this error Call to a member function sync() on null

i am using laravel 10 and spatie 5.8^

if any one have solution pls let me know

**Comment by parallels999** [comment]
With laravel 10, with don't you upgrade to spatie 5.10(current version)?

> Call to a member function sync() on null

I should post some code to have some idea of ​​what you are doing wrong, it seems to me that it is not finding the role, try to delete the spatie cache

**Reply by ravitninjatech** [reply]
i did delete cach but same issue 
![image](https://github.com/spatie/laravel-permission/assets/135598852/bd504453-1b77-4dc7-b022-5582475a2566)

**Reply by parallels999** [reply]
Try current version or dev `6.x-dev` version, maybe that issue has already been solved

**Reply by ravitninjatech** [reply]
base on website 5.8 is latest one 
![image](https://github.com/spatie/laravel-permission/assets/135598852/97c7e559-64db-46da-8e96-511e8e5db8b6)

**Reply by parallels999** [reply]
https://github.com/spatie/laravel-permission/releases
https://github.com/spatie/laravel-permission/tags
https://github.com/spatie/laravel-permission/compare/main...5.10.1

`^5.8` includes all higher versions, see the `^`

**Comment by drbyte** [comment]
Your application code, such as in your User model, probably has some method names or properties that are conflicting.

**Reply by parallels999** [reply, marked as answer]
[content-must-not-have-a-role-or-roles-property-nor-a-roles-method](https://spatie.be/docs/laravel-permission/v5/prerequisites#content-must-not-have-a-role-or-roles-property-nor-a-roles-method)

**Reply by ravitninjatech** [reply]
solved
